shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm

```diff
@@ -17,29 +18,53 @@
 #include "flutter/shell/common/switches.h"
 #include "flutter/shell/common/thread_host.h"
 #include "flutter/shell/platform/darwin/common/command_line.h"
+#include "flutter/shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h"
 #include "flutter/shell/platform/darwin/ios/framework/Source/FlutterDartProject_Internal.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.h"
+#include "flutter/shell/platform/darwin/ios/headless_platform_view_ios.h"
+#include "flutter/shell/platform/darwin/ios/platform_view_ios.h"
 #include "lib/fxl/functional/make_copyable.h"
 
-static std::unique_ptr<shell::PlatformView> CreateHeadlessPlatformView(shell::Shell& shell) {
-  return std::make_unique<shell::PlatformView>(shell, shell.GetTaskRunners());
+static std::unique_ptr<shell::HeadlessPlatformViewIOS> CreateHeadlessPlatformView(
+    shell::Shell& shell) {
+  return std::make_unique<shell::HeadlessPlatformViewIOS>(shell, shell.GetTaskRunners());
 }
 
 static std::unique_ptr<shell::Rasterizer> CreateHeadlessRasterizer(shell::Shell& shell) {
   return std::make_unique<shell::Rasterizer>(shell.GetTaskRunners());
 }
 
+static std::string CreateShellLabel() {
+  static size_t count = 1;
+  std::stringstream stream;
+  stream << "io.flutter.headless.";
+  stream << count++;
+  return stream.str();
+}
+
 @implementation FlutterHeadlessDartRunner {
   shell::ThreadHost _threadHost;
   std::unique_ptr<shell::Shell> _shell;
 }
 
-- (void)runWithEntrypoint:(NSString*)entrypoint {
+- (instancetype)init {
```

Comment:
Does this function and dealloc below need to be added explicitly?

Reply:
No we don't. There was some other functionality added there before, but I forgot to remove these during cleanup. Will remove.